pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -81,86 +83,80 @@ var (
 
 // Reconciler implements controller.Reconciler for Configuration resources.
 type Reconciler struct {
-	*reconciler.Base
+
+	// KubeClientSet allows us to talk to the k8s for core APIs
+	KubeClientSet kubernetes.Interface
+
+	// TriggersClientSet allows us to configure triggers objects
+	TriggersClientSet triggersclientset.Interface
+
+	// CachingClientSet allows us to instantiate Image objects
+	CachingClientSet cachingclientset.Interface
+
 	// listers index properties about resources
 	eventListenerLister listers.EventListenerLister
-	systemNamespace     string
+
+	// systemNamespace is the namespace where the reconciler is deployed
+	systemNamespace string
 }
 
-// Check that our Reconciler implements controller.Reconciler
-var _ controller.Reconciler = (*Reconciler)(nil)
+var (
+	// Check that our Reconciler implements eventlistenerreconciler.Interface
+	_ eventlistenerreconciler.Interface = (*Reconciler)(nil)
+	// Check that our Reconciler implements eventlistenerreconciler.Finalizer
+	_ eventlistenerreconciler.Finalizer = (*Reconciler)(nil)
+)
 
 // Reconcile compares the actual state with the desired, and attempts to
 // converge the two.
-func (c *Reconciler) Reconcile(ctx context.Context, key string) (returnError error) {
-	c.Logger.Infof("event-listener-reconcile %s", key)
-	// Convert the namespace/name string into a distinct namespace and name
-	namespace, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		c.Logger.Errorf("invalid resource key: %s", key)
-		return nil
-	}
-
-	// Get the EventListener resource
-	original, err := c.eventListenerLister.EventListeners(namespace).Get(name)
-	if errors.IsNotFound(err) {
-		// The resource no longer exists, in which case we stop processing.
-		c.Logger.Infof("EventListener %q in work queue no longer exists", key)
-		cfgs, err := c.eventListenerLister.EventListeners(namespace).List(labels.Everything())
-		if err != nil {
-			return err
-		}
-		if len(cfgs) > 0 || namespace == c.systemNamespace {
-			return nil
-		}
-		err = c.KubeClientSet.CoreV1().ConfigMaps(namespace).Delete(eventListenerConfigMapName, &metav1.DeleteOptions{})
-		if errors.IsNotFound(err) {
-			return nil
-		}
-		return err
-	} else if err != nil {
-		c.Logger.Errorf("Error retrieving EventListener %q: %s", name, err)
-		return err
-	}
-
-	// Don't modify the informer's copy
-	el := original.DeepCopy()
+func (r *Reconciler) ReconcileKind(ctx context.Context, el *v1alpha1.EventListener) pkgreconciler.Event {
 	// Initial reconciliation
 	if equality.Semantic.DeepEqual(el.Status, v1alpha1.EventListenerStatus{}) {
```

Comment:
this will never happen with the genreconciler, there is a pre-step (you can opt out) that will call InitializeConditions for you and it marks the generation after you return. It also will expose an error if you reconcile and do nothing (you need to call a mark method)

Reply:
Interesting, I was following the generated stub: https://github.com/tektoncd/triggers/pull/733/files#diff-2518496f4b5a085db3dc51c69af781c2R52

Is there a newer version which calls the IntializeConditions? (This is the older version of genreconciler I think, before leader elections etc.)

Reply:
fixed. calling InitializeConditions in the ReconcileKind each time now